lightning/src/ln/chan_utils.rs

```diff
@@ -1429,12 +1429,14 @@ impl<'a> TrustedCommitmentTransaction<'a> {
 
 		let htlc_redeemscript = get_htlc_redeemscript_with_explicit_keys(&this_htlc, self.opt_anchors(), &keys.broadcaster_htlc_key, &keys.countersignatory_htlc_key, &keys.revocation_key);
 
+		let sighashtype = if self.opt_anchors() { SigHashType::SinglePlusAnyoneCanPay } else { SigHashType::All };
```

Comment:
@TheBlueMatt I'm trying to figure out if the transaction this function is constructing the same transaction as the one constructed in `onchaintx._handler.generate_claim_tx`?